homeassistant/components/gree/climate.py

```diff
@@ -97,11 +101,19 @@
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Gree HVAC device from a config entry."""
-    async_add_entities(
-        [
-            GreeClimateEntity(coordinator)
-            for coordinator in hass.data[DOMAIN][COORDINATOR]
-        ]
+
+    @callback
+    def init_device(coordinator):
+        """Register the device."""
+        async_add_entities([GreeClimateEntity(coordinator)])
+
+    for coordinator in hass.data[DOMAIN][COORDINATORS]:
+        init_device(coordinator)
+
+    hass.data[DOMAIN].setdefault(DISPATCHERS, [])
```

Comment:
Move this to `async_setup_entry` in `__init__.py`

Reply:
Same needed in switch.py, done.